lib/git-lfs-s3/application.rb

```
@@ -83,11 +93,15 @@ def protected!
       status service.status
       body MultiJson.dump(service.response)
     end
-
+    
     post '/verify', provides: 'application/vnd.git-lfs+json' do
       data = MultiJson.load(request.body.tap { |b| b.rewind }.read)
       object = object_data(data['oid'])
-
+      if settings.public_server and settings.ceph_s3
+        if object.exists? and not object.acl.grants.include?(public_read_grant)
+          object.acl.put(acl: "public-read")
```

I agree. Now I fail early and the remaining logic is simpler.